mender-update/daemon/state_test.cpp

```diff
@@ -1067,9 +1195,31 @@ vector<StateTransitionsTestCase> GenerateStateTransitionsTestCases() {
 			.state_chain =
 				{
 					"Download_Enter_00",
+					"Download_Enter_00",
+					"Download",
+					"Download_Leave_00",
+					"ArtifactInstall_Enter_00",
+					"ArtifactInstall",
+					"ArtifactInstall_Leave_00",
+					"ArtifactReboot_Enter_00",
+					"ArtifactReboot",
+					"ArtifactVerifyReboot",
+					"ArtifactReboot_Leave_00",
+					"ArtifactCommit_Enter_00",
+					"ArtifactCommit",
+					"ArtifactCommit_Leave_00",
+					"Cleanup",
```

Comment:
This is wrong, it should not run any installation steps.

Reply:
Hmm, but when `Download_Enter_00` is rerun after having been killed, it succeeds, how can it then know it should not continue down the successful path?

Reply:
So this is literally what we want?

```
					"Download_Enter_00",
					"Cleanup",
```

Reply:
I think that cleanup might be missing from the record because it fails so early, so the Update Module isn't initialized yet. It still runs internally, but you probably won't get it in the list. You'll have to see.

Reply:
Ur right! Was looking into why it wasn't showing just now. Got tired, and checked my mail :laughing: